### PR TITLE
Lookout cloud security Solution changes

### PR DESCRIPTION
**Pull request by VenkataPrathibha**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by v-mchatla** [comment]
Hi @devikamehra,
Can you please review the changes and share your feedback.
Thanks

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py
Comment (anki-narravula) [inline Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py:35]: I think, if customer choose None, should we not process historical data?

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py
Comment (anki-narravula) [inline Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py:106]: 1. I think we need to remove this statement. We should post only if we don't see any exceptions. In case of exceptions, time should not get incremented.

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py
Comment (anki-narravula) [inline Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py:230]: 2000 can be a variable

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py
Comment (anki-narravula) [inline Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py:227]: Not considering the scenario, what happens when we are able to send only 1-2 batches gen_chunks()  and remaining got failures. We should increment, only when we are able to send data successfully to sentinel.

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/function.json
Comment (anki-narravula) [inline Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/function.json:8]: This is very good idea to consider as a configurable parameter

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/azuredeploy_LookoutCloudSecurityConnector_FunctionApp.json
Comment (anki-narravula) [inline Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/azuredeploy_LookoutCloudSecurityConnector_FunctionApp.json:36]: This can be more descriptive and some example

File: .vscode/settings.json
Comment (anki-narravula) [inline .vscode/settings.json:15]: Is this file required to check-in?

File: .vscode/launch.json
Comment (anki-narravula) [inline .vscode/launch.json:13]: Is this file required to check-in?

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/azuredeploy_LookoutCloudSecurityConnector_FunctionApp.json
Comment (anki-narravula) [inline Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/azuredeploy_LookoutCloudSecurityConnector_FunctionApp.json:201]: After merge, we need to change the target to master branch

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityConnector_API_FunctionApp.json
Comment (anki-narravula) [inline Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityConnector_API_FunctionApp.json:106]: After merge, we need to change the target to master branch

**Review by anki-narravula — COMMENTED** [review]
Hi @VenkataPrathibha , I have reviewed and added my comments. Reviewed only 1 function, and the comments applies to rest of 2 functions as well.

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py
Comment (VenkataPrathibha) [reply Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py]: done.

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py
Comment (VenkataPrathibha) [reply Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py]: done

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/azuredeploy_LookoutCloudSecurityConnector_FunctionApp.json
Comment (VenkataPrathibha) [reply Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/azuredeploy_LookoutCloudSecurityConnector_FunctionApp.json]: As discussed provide instruction that the url must start with https://

File: .vscode/settings.json
Comment (VenkataPrathibha) [reply .vscode/settings.json]: Reverted.

File: .vscode/launch.json
Comment (VenkataPrathibha) [reply .vscode/launch.json]: Reverted

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/azuredeploy_LookoutCloudSecurityConnector_FunctionApp.json
Comment (VenkataPrathibha) [reply Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/azuredeploy_LookoutCloudSecurityConnector_FunctionApp.json:204]: Sure

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py
Comment (VenkataPrathibha) [reply Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityAnamolies/__init__.py]: As discussed applied default values

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityConnector_API_FunctionApp.json
Comment (VenkataPrathibha) [reply Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityConnector_API_FunctionApp.json:106]: Sure

File: Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py
Comment (VenkataPrathibha) [reply Solutions/Lookout Cloud Security/Data Connectors/LookoutCSConnector/LookoutCloudSecurityEvents/__init__.py:221]: Rewrote the entire chunk process in order to capture the latest timestamp and tested. Please review.

**Review by VenkataPrathibha — COMMENTED** [review]
Hi @anki-narravula , I have updated the changes and provided my comments. Please review and share your inputs. Thanks!

**Review by anki-narravula — APPROVED** [review]
Verified functionality of the data connector and looks good. Can be merged after other regular checks for the given content type

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**
